### DISCUSSION TITLE
About options when converting png to psd

### DISCUSSION THREAD

**Discussion by la-darknesss** [ImageMagick/ImageMagick#6201] [Help] [answered]
### ImageMagick version

7.1.0-Q16-HDRI

### Operating system, version and so on

windows10

### Description

The following command can be used to add a png file to a psd layer and create a psd file.
`magick.exe png_01.png png_02.png ( -clone 0,1 -flatten ) -insert 0 output_psd.psd`

However, the dpi of the added psd layer is 0.
Is it possible to change it to the dpi or ppi of the png file that is being converted from? 

Thank you in advance.

**Comment by fmw42** [comment]
Imagemagick usually uses the information from the first image in the command line for the output. 

Post example images, if that is not true for your situation.

Which is the added image in your case above?

**Reply by la-darknesss** [reply]
Thank you for your response.

For example, the attached image is 72 dpi.
However, when converted to psd, PhotoShop shows it as 0 ppi.
`magick.exe jpg_01.jpg output_psd.psd`
![jpg_01](https://user-images.githubusercontent.com/129155371/228582247-ac954fbe-0f58-412a-b7e1-2b2bae3ac0f5.jpg)

**Comment by fmw42** [comment]
Neither Imagemagick identify -verbose nor EXIFTOOL show any density as far as I can see.  How do you know that it has a density of 96 dpi. If Imagemagick does not see that density then it cannot make a PDF with that density.

**Reply by la-darknesss** [reply]
Sorry, the correct dpi was 72 dpi.
The way I checked was to load jpg_01.jpg in PhotoShop and check it from the properties.

I see that currently Imagemagick does not recognize the dpi.
I will check the status of jpg_01.jpg from Imagemagick.
By the way, is it possible to change the ppi to a specified number?

**Reply by fmw42** [reply]
`convert image.jpg -density XX image.jpg` will set the density.  The default density is 72, which is what Imagemagick uses. But density is only needed for printing and if not set will default to 72 as far as I know.

**Comment by fmw42** [comment]
Looks like a possible issue with Imagemagick.  I guess Imagemagick does not see any defined density and so either leaves it blank or sets it to 0.  Photoshop must assume it is 72 when not specified. So what you need to do when converting from JPG to PSD is to set a density.

`convert image.jpg -density 72 image.psd`

Use any density that you want.

**Reply by la-darknesss** [reply]
The reason why Imagemagick does not recognize it may be what snibgo is talking about.
However, I tried the above on the command line and was able to set it to 72ppi.
Thank you very much.

Sorry, it is added in a layer locked state.
Is it possible to unlock the layer lock?

**Comment by snibgo** [comment]
> For example, the attached image is 72 dpi.

No, it isn't.

> Looks like a possible issue with Imagemagick.

I don't think so. The image of the masked girl, according to exiftool, has:
```
Resolution Unit                 : None
X Resolution                    : 1
Y Resolution                    : 1
```
So it has no resolution.

**Reply by la-darknesss** [reply]
Thank you for your response.
I see, the resolution does not exist.
The attached image is the value I checked in PhotoShop.
What is different from the resolution in exiftool?
![photoshop_r](https://user-images.githubusercontent.com/129155371/228798317-925cb8b2-1825-4dcf-b6ce-2600ab43bb8d.png)

**Comment by snibgo** [comment, marked as answer]
`exiftool` tells us the metadata that is actually in the file.

I don't use Photoshop, but I suppose it tells us the metadata of the image *after* it has been read into memory, and PS has used default values for any missing metadata.

If I name your image `tigerGirl.jpg` and use IM like this:
```
magick tigerGirl.jpg -density 99 x.png
```
... then `exiftool x.png` says this:
```
Pixels Per Unit X               : 99
Pixels Per Unit Y               : 99
Pixel Units                     : Unknown
```
So that is still bad. We should specify the units:
```
magick tigerGirl.jpg -density 99 -units pixelsperinch x.png
```
Then `exiftool x.png` says this:
```
Pixels Per Unit X               : 3897
Pixels Per Unit Y               : 3897
Pixel Units                     : meters
```
One meter is 1000/25.4 = about 39.37008 inches, so 3897 pixels per meter is 3897/39.37008 = 98.9838 pixels per inch.

**Reply by la-darknesss** [reply]
Thank you for the detailed information.
I see, 72 ppi may be PhotoShop's default value.
So you need to set the value with `-density`.
Thanks for the reference.

**Comment by la-darknesss** [comment]
Sorry, I have one last question.
By default, the layers are locked.
Is there any way to unlock the layer lock?

**Reply by fmw42** [reply]
The command

`magick 228582247-ac954fbe-0f58-412a-b7e1-2b2bae3ac0f5.jpg -density 99 -units pixelsperinch x.jpg
`
creates one (background) layer which you cannot turn off. It is the same as for the input.

If you create multiple layers, I think they will all be checked by default and I know of no way to disable them from Imagemagick.  Perhaps an IM developer might comment further or correct me.

**Reply by la-darknesss** [reply]
Understood.
It seems that the layer lock setting cannot be changed from ImageMagick.
Thank you for confirming this.
